### DISCUSSION TITLE
Problem with YAML syntax

### DISCUSSION THREAD

**Discussion by lpzhang-code** [community/community#25495] [Actions]
<p>Hello everyone!</p>
<p>I am writing my workflow file on YAML but it says there is an error on line 4</p>
<pre><code class="lang-auto">              if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
                  echo "::set-env name=COMPOSE_SUFFIX::staging"
                  echo "::set-env name=PROJECT::clerk_test"
              elif [ "${{ github.ref }}" = "refs/heads/master" ]; then
                  echo "::set-env name=COMPOSE_SUFFIX::prod"
                  echo "::set-env name=PROJECT::clerk"
              else
                  echo "Please select develop (staging) or master (prod) branch"
                  exit 1
              fi
</code></pre>
<p>Can anybody tell me what is wrong with my syntax?</p>

**Comment by thomasjo** [comment]
<p>Hey <a class="mention" href="/lpzhang5">@lpzhang5</a>, can you share the full YAML file? Either inline or a link to your repository. It would make it easier for us to assist you. The reason I want some more context is that the code you have shared is not YAML, it’s a Bash script. The problem you are facing might be related to how you are embedding the Bash script in the YAML file, but without more context, I can only guess.</p>

**Comment by lpzhang-code** [comment]
<p>Hello <a class="mention" href="/thomasjo">@thomasjo</a> ,</p>
<p>The full YAML file is here</p>
<p>there is some bash script embedded for setting env vars, it says there is problem at the line for ‘elif’</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/lpzhang5/CI-Workflow/blob/abbdda05cac4326dd757a0de6ad0095f98b81be5/.github/workflows/deploy.yml" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/lpzhang5/CI-Workflow/blob/abbdda05cac4326dd757a0de6ad0095f98b81be5/.github/workflows/deploy.yml" target="_blank" rel="noopener nofollow ugc">lpzhang5/CI-Workflow/blob/abbdda05cac4326dd757a0de6ad0095f98b81be5/.github/workflows/deploy.yml</a></h4>
<pre><code class="lang-yml">name: Deploy Application
on: [workflow_dispatch]
jobs:
    deploy:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v2
            - name: Set variables
              run: echo "Playing around"</code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by thomasjo** [comment, marked as answer]
<p>Thanks for that. The problem is caused by missing indentation in the multiline block for the Bash script. Add indentation to the entire Bash script block and your problem should be resolved.</p>
<pre><code class="lang-yaml">name: Deploy Application
on: [workflow_dispatch]
jobs:
    deploy:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v2
            - name: Set variables
              run: |
                if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
                    echo "::set-env name=COMPOSE_SUFFIX::staging"
                    echo "::set-env name=PROJECT::clerk_test"
                elif [ "${{ github.ref }}" = "refs/heads/master" ]; then
                    echo "::set-env name=COMPOSE_SUFFIX::prod"
                    echo "::set-env name=PROJECT::clerk"
                else
                    echo "Please run with master (prod) or develop (staging) branch"
                    exit 1
                fi
</code></pre>

**Reply by LilyAnderssonLee** [reply]
Hi, I am having the similar issue in my .github/workflows/ci.yml. Here is my [PR](https://github.com/nf-core/taxprofiler/pull/350)
Error:  
`Run Wandalen/wretry.action@v1.0.11
sh: Syntax error: end of file unexpected (expecting "fi")`

Thanks in advance!

**Comment by lpzhang-code** [comment]
<p>Thank you!</p>
<p>It is fixed now. Quite a few headaches due to this minor but subtle problem!</p>
